translator/translate/otel/processor/resourceprocessor/translator.go

```diff
@@ -65,6 +69,57 @@ func (t *translator) Translate(conf *confmap.Conf) (component.Config, error) {
 	c := confmap.NewFromStringMap(map[string]any{
 		"attributes": attributes,
 	})
+
+	if strings.HasPrefix(t.Name(), common.PipelineNameJmx) {
+		attributes = t.getJMXAttributes(conf)
+	} else if t.Name() == common.PipelineNameContainerInsightsJmx {
+		clusterName, ok := common.GetString(conf, common.ConfigKey(eksKey, "cluster_name"))
+
+		if ok {
+			c = confmap.NewFromStringMap(map[string]any{
+				//from config
+				"attributes": []any{
+					map[string]any{
+						"key":            "Namespace",
+						"from_attribute": "k8s.namespace.name",
+						"action":         "insert",
+					},
+					map[string]any{
+						"key":    "ClusterName",
+						"value":  clusterName,
+						"action": "upsert",
+					},
+					map[string]any{
+						"key":            "NodeName",
+						"from_attribute": "host.name",
+						"action":         "insert",
+					},
+				},
+			})
```

Comment:
The change isn't quite right. We shouldn't be calling any of this after we create the confmap. The idea is to populate the `var attributes []any` before we create the confmap on line 69.

Reply:
Ahh got it have the confmap be populated by attributes